stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1399,6 +1425,45 @@ func _TFCOpSetAttrString(_ op: CTFEOp,
   }
 }
 
+/// Wrapper around TFE_OpSetAttrStringList that handles converting the Swift
+/// Strings into buffers that TFE_OpSetAttrStringList can read.
+@usableFromInline
+@_silgen_name("_swift_tfc_OpSetAttrStringArray")
+func _TFCOpSetAttrStringArray(_ op: CTFEOp,
+                             _ attrName: UnsafePointer<Int8>,
+                             _ strings: [String]) {
+  // Collect all the strings' utf8 bytes into a single array so that we can
+  // address all the strings with a single
+  // `flattenedStringBytes.withUnsafeBufferPointer`.
+  var flattenedStringBytes: [CChar] = []
+  var lengths: [Int] = []
+  for string in strings {
+    // Don't include the null-terminator because TFE_OpSetAttrStringList uses
+    // lengths instead of null-terminators.
+    let stringBytes = string.utf8CString.dropLast()
```

Comment:
is there a single way to assert the last byte is indeed `\0`?

Reply:
It's guaranteed by the `utf8CString` API: https://developer.apple.com/documentation/swift/string/2430818-utf8cstring, so assertion isn't strictly necessary. But i'm fine either way.

Reply:
It's guaranteed by the `utf8CString` API: https://developer.apple.com/documentation/swift/string/2430818-utf8cstring, so assertion isn't strictly necessary. But i'm fine either way.